File: doc-examples/example-jsonb-java/src/main/java/example/MilesAdapter.java
package example;

import io.micronaut.context.annotation.Requires;
import jakarta.annotation.Priority;
import jakarta.inject.Singleton;
import jakarta.json.bind.adapter.JsonbAdapter;

@Singleton
@Requires(property = "spec.name", value = "jsonb-extension-beans")
@Priority(10)
public final class MilesAdapter implements JsonbAdapter<Miles, String> {
    @Override
    public String adaptToJson(Miles obj) {
        return obj.getValue() + " mi";
    }

    @Override
    public Miles adaptFromJson(String obj) {
        return new Miles(Integer.parseInt(obj.replace(" mi", "")));
    }
}

File: doc-examples/example-jsonb-python/src/main/python/example/MilesAdapter.py
from jakarta.annotation import Priority
from jakarta.inject import Singleton
from jakarta.json.bind.adapter import JsonbAdapter
from micronaut.context.annotation import Requires

from example.Miles import Miles


@Singleton
@Requires(property="spec.name", value="jsonb-extension-beans")
@Priority(10)
class MilesAdapter(JsonbAdapter[Miles, str]):
    def adaptToJson(self, obj: Miles) -> str:
        return f"{obj.value} mi"

    def adaptFromJson(self, obj: str) -> Miles:
        return Miles(int(obj.replace(" mi", "")))
